multierror.go

```diff
@@ -49,3 +50,71 @@ func (e *Error) GoString() string {
 func (e *Error) WrappedErrors() []error {
 	return e.Errors
 }
+
+// Unwrap returns an error from Error (or nil if there are no errors).
+// This error returned will further support Unwrap to get the next error,
+// etc. The order will match the order of Errors in the multierror.Error
+// at the time of calling.
+//
+// The resulting error supports errors.As/Is/Unwrap so you can continue
+// to use the stdlib errors package to introspect further.
+//
+// This will perform a shallow copy of the errors slice. Any errors appended
+// to this error after calling Unwrap will not be available until a new
+// Unwrap is called on the multierror.Error.
+func (e *Error) Unwrap() error {
+	// If we have no errors then we do nothing
+	if e == nil || len(e.Errors) == 0 {
+		return nil
+	}
+
+	// Shallow copy the slice
+	errs := make([]error, len(e.Errors))
+	copy(errs, e.Errors)
+
+	return &chain{errors: errs}
+}
+
+// chain implements the interfaces necessary for errors.Is/As/Unwrap to
+// work in a deterministic way with multierror. A chain tracks a list of
+// errors while accounting for the current represented error. This lets
+// Is/As be meaningful.
+//
+// Unwrap returns the next error. In the cleanest form, Unwrap would return
+// the wrapped error here but we can't do that if we want to properly
+// get access to all the errors. Instead, users are recommended to use
+// Is/As to get the correct error type out.
+type chain struct {
+	idx    int
+	errors []error
+}
+
+// Error implements the error interface
+func (e *chain) Error() string {
+	return e.current().Error()
+}
+
+// Unwrap implements errors.Unwrap by returning the next error in the
+// chain or nil if there are no more errors.
+func (e *chain) Unwrap() error {
+	next := e.idx + 1
+	if len(e.errors) <= next {
+		return nil
+	}
+
+	return &chain{idx: next, errors: e.errors}
+}
+
+// As implements errors.As by attempting to map to the current value.
+func (e *chain) As(target interface{}) bool {
+	return errors.As(e.current(), target)
+}
+
+// Is implements errors.Is by comparing the current value directly.
+func (e *chain) Is(target error) bool {
+	return e.current() == target
```

Comment:
The Go docs for [errors.Is](https://golang.org/pkg/errors/#Is) say:

> An error is considered to match a target if it is equal to that target or if it implements a method Is(error) bool such that Is(target) returns true.

IMO it is therefor reasonable to expect that same logic is used for errors wrapped by `go-multierror`, i.e. if a wrapped error implements `Is(error) bool`, then that method should be called to determine equality rather than a simple `==` check.

This should fix it:

```suggestion
	return errors.Is(e.current(), target)
```